### DISCUSSION TITLE
Downloading Media

### DISCUSSION THREAD

**Discussion by ltalreja-gryphon** [lhz516/react-h5-audio-player#115] [Q&A] [answered]
Is there a way you can trigger audio file download?

**Comment by lhz516** [comment, marked as answer]
You can add a custom button on the player to make it trigger download

See advanced custom layout https://static.hanzluo.com/react-h5-audio-player-storybook/index.html?path=/docs/layouts-advanced--stacked

**Reply by ltalreja-gryphon** [reply]
I tried doing that, however, it opens the audio file in a new tab with the default media player of the browser because the hosts where the front-end is served and the audio file is located are different. With the default media player of the browser, this wasn't the case, you could directly download the file.

![image](https://user-images.githubusercontent.com/85502721/133794192-6d75935c-278f-4ab9-a1b8-c965126414d0.png)
![image](https://user-images.githubusercontent.com/85502721/133794217-d98c1aa3-eab6-47fe-b2e0-ce5e30abf3ae.png)

**Reply by lhz516** [reply]
You may use `<a href="/path/song.mp3" download>Download</a>`. However, this only works for relative path. If it's absolute path which includes http prefix, it will still open a new window with default audio player,

When you go to https://filesamples.com/formats/mp3 and inspect the download element. you got
```html
<a href="/samples/audio/mp3/Symphony No.6 (1st movement).mp3" download="" class="btn btn-primary pull-right"><i class="fas fa-download"></i> Download</a>
```
The download works as expected. In your own website, if you use `https://filesamples.com/samples/audio/mp3/Symphony No.6 (1st movement).mp3` as `href`. It still opens default player when you click it.

**Reply by ltalreja-gryphon** [reply]
So, do you suggest downloading and saving the audio file on the backend server and adding this path to the anchor component?

**Reply by lhz516** [reply]
I would upload the file to cloud storage and use a proxy to make it a relative path.
